src/native/fs/fs_napi.cpp

```diff
@@ -1350,7 +1350,7 @@ struct FileWrap : public Napi::ObjectWrap<FileWrap>
             LOG("FS::FileWrap::dtor: file not closed " << DVAL(_path) << DVAL(_fd));
             int r = ::close(_fd);
             if (r) LOG("FS::FileWrap::dtor: file close failed " << DVAL(_path) << DVAL(_fd) << DVAL(r));
-            _fd = 0;
+            _fd = -1;
```

Comment:
need to fix the condition in the dtor from if (_fd) to if(_fd >= 0) ...
Also fix the same condition in AutoCloser.
Also I would add a check to FileClose, so if (_fd >= 0) it can close it, otherwise ignore (already closed or not opened).
Also I see that FileReplacexattr is missing the CHECK_WRAP_FD(fd) call.
Also I see that LinkFileAt and UnlinkFileAt initialize _delete_fd/_replace_fd to 0 and better initialize to -1 and check accordingly.